src/xdp-main.c

```diff
@@ -0,0 +1,249 @@
+{
+  XdpMain *xdp_main = user_data;
+  g_autoptr(GError) error = NULL;
+
+  if (!xdp_context_register (xdp_main->context, connection, &error))
+    {
+      g_critical ("Starting portals failed: %s", error->message);
+      xdp_main_exit (xdp_main, 1);
+    }
+}
+
+static void
+on_name_acquired (GDBusConnection *connection,
+                  const gchar     *name,
+                  gpointer         user_data)
+{
+  g_debug ("%s acquired", name);
+}
+
+static void
+on_name_lost (GDBusConnection *connection,
+              const gchar     *name,
+              gpointer         user_data)
+{
+  XdpMain *xdp_main = user_data;
+
+  xdp_main_exit (xdp_main, 0);
+  g_debug ("Terminated because dbus name was lost.");
+}
+
+static gboolean
+on_signal (gpointer user_data)
+{
+  XdpMain *xdp_main = user_data;
+
+  xdp_main_exit (xdp_main, 0);
+  g_debug ("Terminated with signal.");
+
+  return G_SOURCE_REMOVE;
+}
+
+int
+main (int argc, char *argv[])
+{
+  XdpMain xdp_main;
+  g_autoptr(GMainLoop) loop = NULL;
+  g_autoptr(XdpContext) context = NULL;
+  guint owner_id;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GDBusConnection) session_bus = NULL;
+  g_autoptr(GSource) signal_handler_source = NULL;
+  g_autoptr(GOptionContext) option_context = NULL;
+
+  gboolean opt_verbose;
+  gboolean opt_replace;
+  gboolean opt_show_version;
+
+  GOptionEntry entries[] = {
+    { "verbose", 'v', 0, G_OPTION_ARG_NONE, &opt_verbose, "Print debug information during command processing", NULL },
+    { "replace", 'r', 0, G_OPTION_ARG_NONE, &opt_replace, "Replace a running instance", NULL },
+    { "version", 0, 0, G_OPTION_ARG_NONE, &opt_show_version, "Show program version.", NULL},
+    { NULL }
+  };
+
+  if (g_getenv ("XDG_DESKTOP_PORTAL_WAIT_FOR_DEBUGGER") != NULL)
+    {
+      g_printerr ("\ndesktop portal (PID %d) is waiting for a debugger. "
+                  "Use `gdb -p %d` to connect. \n",
+                  getpid (), getpid ());
+
+      if (raise (SIGSTOP) == -1)
+        {
+          g_printerr ("Failed waiting for debugger\n");
+          exit (1);
+        }
+
+      raise (SIGCONT);
+    }
+
+  setlocale (LC_ALL, "");
+  bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+  bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+  textdomain (GETTEXT_PACKAGE);
+
+  /* Note: if you add any more environment variables here, update
+   * handle_launch() in dynamic-launcher.c to unset them before launching apps
+   */
+  /* Avoid even loading gvfs to avoid accidental confusion */
+  g_setenv ("GIO_USE_VFS", "local", TRUE);
+
+  /* Avoid pointless and confusing recursion */
+  g_unsetenv ("GTK_USE_PORTAL");
+
+  option_context = g_option_context_new ("- desktop portal");
+  g_option_context_set_summary (option_context,
+      "A portal service for flatpak and other desktop containment frameworks.");
+  g_option_context_set_description (option_context,
+      "xdg-desktop-portal works by exposing D-Bus interfaces known as portals\n"
```

Comment:
Wrong indentation.

Reply:
What's wrong about it? If the opening bracket of the function call is too far to the right that the arguments all exceed the line length limit, one has to put them there.